packages/stream/test/test-pump.js

```diff
@@ -0,0 +1,323 @@
+// @ts-check
+/* global process */
+
+import 'ses';
+import './lockdown.js';
+
+import rawTest from 'ava';
+import { wrapTest } from '@endo/ses-ava';
+import { pump, prime } from '../index.js';
+
+const test = wrapTest(rawTest);
+
+test('pump happy path', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  t.plan(2);
+
+  async function* source() {
+    for (let i = 0; i < 4; i += 1) {
+      yield i;
+    }
+    yield* [4, 5, 6];
+    return '.';
+  }
+
+  async function* target() {
+    const collected = [];
+    try {
+      for (;;) {
+        collected.push(yield);
+      }
+    } catch (error) {
+      t.fail(error);
+    } finally {
+      t.assert(true);
+      t.deepEqual(collected, [0, 1, 2, 3, 4, 5, 6]);
+    }
+  }
+
+  await pump(prime(target()), source());
+});
+
+test('pump target closes early', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  t.plan(5);
+
+  async function* source() {
+    for (let i = 0; i < 10; i += 1) {
+      yield i;
+    }
+    return 'X';
+  }
+
+  async function* target() {
+    for (let i = 0; i < 5; i += 1) {
+      t.is(yield, i);
+    }
+    return 'Y';
+  }
+
+  await pump(prime(target()), source());
+});
+
+test('pump with error', async (/** @type {import('ava').ExecutionContext} */ t) => {
+  t.plan(3);
+
+  async function* source() {
+    for (let i = 0; i < 4; i += 1) {
+      yield i;
+    }
+    yield* [4, 5, 6];
+    throw new Error('Abort');
+  }
+
+  const collected = [];
+
+  async function* target() {
+    try {
+      for (;;) {
+        collected.push(yield);
+      }
+    } catch (error) {
+      t.is(error.message, 'Abort');
+      t.deepEqual(collected, [0, 1, 2, 3, 4, 5, 6]);
+    } finally {
+      t.assert(true);
+    }
+  }
+
+  try {
+    await pump(prime(target()), source());
+  } catch (error) {
+    // Between Node.js 12 and 14, V8 stopped propagating the error from the
```

Comment:
ugh, that's sad

Reply:
What does this mean? I don't understand the comment. What did node stop propagating?

Reply:
I will change the comment to this prose:
```
    // In Node.js 12, the error thrown on the last line of the `source`
    // generator does not propagate to the promise returned by `pump`.
    // In Node.js 14 and 16, the error does propagate.
```

Reply:
Ok I re-read this, and now I'm thinking this was a bug in older v8 versions that was fixed? I find this error swallowing fairly surprising.